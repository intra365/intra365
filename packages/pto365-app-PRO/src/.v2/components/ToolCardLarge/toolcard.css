.ToolCard{
    width: 100%;
}
.ToolCardHeader {
   
    cursor: pointer;
   
    height: 48px;
    display: flex;
    flex-direction: row
}

.ToolCardIcon{
margin: 8px 8px;   

 
}

.ToolCardTitle{
  
    line-height: 48px
}
.ToolCardJump{
   
    margin-top: 8px;
    align-self: flex-start;
}

.ToolCardDetails {
    margin: 0px;
    min-height: 50px;
    flex-direction: row
}
.ToolCardText {
    margin:5px;
    
    
}

.ToolCardReadmoreLink {
    margin-top: 5px;
    font-size: smaller
}