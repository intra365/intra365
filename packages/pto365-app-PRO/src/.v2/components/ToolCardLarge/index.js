import React, { Component } from 'react';
import PropTypes from 'prop-types'
import AppIconMini from '../AppIconMini'
import ReactPlacerHolder from 'react-placeholder'
import './toolcard.css'
import { DefaultButton } from '../../../../node_modules/office-ui-fabric-react';
import Jumpto365Service from '../../services'
import { Link } from "@reach/router"
/**
 * Describe overall purpose of the component
 *
 * @export
 * @class ToolCardLarge
 * @extends {Component}
 */

 
export default class ToolCardLarge extends Component { 

    static propTypes  = {
        about : PropTypes.string,
        color :   PropTypes.string.isRequired,
        name :   PropTypes.string.isRequired,
        iconUrl :   PropTypes.string.isRequired,
        description : PropTypes.string,
        onClick : PropTypes.func,
        onJump : PropTypes.func,
    }
    
    jumpto365Service = null
    constructor(props) {
        super(props)
        this.state = {ready:false,description:"..."};
    }

    componentDidMount = () => {
        this.jumpto365Service = new Jumpto365Service()
        this.jumpto365Service.getToolDocument(this.props.name).then(document => {
            this.setState({ready:true,title:document.title,description:document.properties && document.properties.inshort ? document.properties.inshort : "",
            hasDocument : !document.isEmpty})
        })
        .catch(err => {
            this.setState({ready: true,description:JSON.stringify(err)})
        })

    }

    onClick = (e) => {
        e.stopPropagation();
        if (this.props.onClick){
            this.props.onClick(this.props.name,this.state)
        }
    }
    
    onJump= (e) => {
        e.stopPropagation();
        if (this.props.onJump){
            this.props.onJump(this.props.name,this.state)
        }
    }
    //TODO: Dim out jump to if no function then implement the Jumpto button
    //TODO: Check scaling

    /**
     * Required method return the output of the component
     *
     * @returns
     * @memberof ToolCardLarge
     */
    render() {


        var description = this.state.hasDocument ?
        <div>
        <div> {this.state.description }</div>
        <a class="ToolCardReadmoreLink" onclick={this.onClick} href="#">Read more</a>
        </div>
        : "No description"
        
        return (
    // https://css-tricks.com/snippets/css/a-guide-to-flexbox/

              <div onClick={this.onClick} className="ToolCard" >
                <div className="ToolCardHeader" style={{ backgroundColor:this.props.color,color:"#ffffff"}}>
                    <div className="ToolCardIcon" >
                    
                            <AppIconMini  onClick={this.onClick}  backgroundColor={this.props.color} name={this.props.name} iconUrl={this.props.iconUrl} />
                    </div>
                    <div className="ToolCardTitle" style={{lineHeight:"48px"}}>
                        <div className="ms-fontSize-xl ms-fontWeight-semilight">
                            {this.state.title}
                        </div>
                    </div>
                          
                    
                   
                </div>
                <div className="ToolCardDetails" >
                    <div className="ms-font-m" >
                    <ReactPlacerHolder type='text' showLoadingAnimation  rows={3} ready={this.state.ready}>

                            {description}


                            </ReactPlacerHolder>

                            {/* <div className="ToolCardJump "  >
                    {this.state.onJump!== null &&
                        <DefaultButton primary={false} onClick={this.state.onJump} target="_blank"  text="Jumpto" />
                        }
                    </div> */}
                        </div>
  
            </div>
           

            
        </div>

        )
    }
}

