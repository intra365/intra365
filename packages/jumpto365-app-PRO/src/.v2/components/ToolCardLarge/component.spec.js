import React, { Component } from 'react';

import ToolCardLarge from '.'


test('Check that it renders', () => {


    const wrapper = shallow(<ToolCardLarge color="#264D8C" name="word" iconUrl="https://jumpto365.com/resources/images/app/Word.png"  />);


    expect(wrapper).toMatchSnapshot();
 
}

)

