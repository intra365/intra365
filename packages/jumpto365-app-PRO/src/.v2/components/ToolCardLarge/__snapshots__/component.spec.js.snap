// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check that it renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ToolCardLarge
    color="#264D8C"
    iconUrl="https://jumpto365.com/resources/images/app/Word.png"
    name="word"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ToolCardHeader"
          style={
                    Object {
                              "backgroundColor": "#264D8C",
                              "color": "#ffffff",
                            }
          }
>
          <div
                    className="ToolCardIcon"
          >
                    <AppIconMini
                              backgroundColor="#264D8C"
                              iconUrl="https://jumpto365.com/resources/images/app/Word.png"
                              name="word"
                              onClick={[Function]}
                    />
          </div>
          <div
                    className="ToolCardTitle"
                    style={
                              Object {
                                        "lineHeight": "48px",
                                      }
                    }
          >
                    <div
                              className="ms-fontSize-xl ms-fontWeight-semilight"
                    />
          </div>
</div>,
        <div
          className="ToolCardDetails"
>
          <div
                    className="ms-font-m"
          >
                    <ReactPlaceholder
                              color="#CDCDCD"
                              delay={0}
                              ready={false}
                              rows={3}
                              showLoadingAnimation={true}
                              type="text"
                    >
                              No description
                    </ReactPlaceholder>
          </div>
</div>,
      ],
      "className": "ToolCard",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="ToolCardIcon"
>
              <AppIconMini
                            backgroundColor="#264D8C"
                            iconUrl="https://jumpto365.com/resources/images/app/Word.png"
                            name="word"
                            onClick={[Function]}
              />
</div>,
            <div
              className="ToolCardTitle"
              style={
                            Object {
                                          "lineHeight": "48px",
                                        }
              }
>
              <div
                            className="ms-fontSize-xl ms-fontWeight-semilight"
              />
</div>,
          ],
          "className": "ToolCardHeader",
          "style": Object {
            "backgroundColor": "#264D8C",
            "color": "#ffffff",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <AppIconMini
                backgroundColor="#264D8C"
                iconUrl="https://jumpto365.com/resources/images/app/Word.png"
                name="word"
                onClick={[Function]}
/>,
              "className": "ToolCardIcon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "backgroundColor": "#264D8C",
                "iconUrl": "https://jumpto365.com/resources/images/app/Word.png",
                "name": "word",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="ms-fontSize-xl ms-fontWeight-semilight"
/>,
              "className": "ToolCardTitle",
              "style": Object {
                "lineHeight": "48px",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "ms-fontSize-xl ms-fontWeight-semilight",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="ms-font-m"
>
            <ReactPlaceholder
                        color="#CDCDCD"
                        delay={0}
                        ready={false}
                        rows={3}
                        showLoadingAnimation={true}
                        type="text"
            >
                        No description
            </ReactPlaceholder>
</div>,
          "className": "ToolCardDetails",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ReactPlaceholder
              color="#CDCDCD"
              delay={0}
              ready={false}
              rows={3}
              showLoadingAnimation={true}
              type="text"
>
              No description
</ReactPlaceholder>,
            "className": "ms-font-m",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "No description",
              "color": "#CDCDCD",
              "delay": 0,
              "ready": false,
              "rows": 3,
              "showLoadingAnimation": true,
              "type": "text",
            },
            "ref": null,
            "rendered": "No description",
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ToolCardHeader"
            style={
                        Object {
                                    "backgroundColor": "#264D8C",
                                    "color": "#ffffff",
                                  }
            }
>
            <div
                        className="ToolCardIcon"
            >
                        <AppIconMini
                                    backgroundColor="#264D8C"
                                    iconUrl="https://jumpto365.com/resources/images/app/Word.png"
                                    name="word"
                                    onClick={[Function]}
                        />
            </div>
            <div
                        className="ToolCardTitle"
                        style={
                                    Object {
                                                "lineHeight": "48px",
                                              }
                        }
            >
                        <div
                                    className="ms-fontSize-xl ms-fontWeight-semilight"
                        />
            </div>
</div>,
          <div
            className="ToolCardDetails"
>
            <div
                        className="ms-font-m"
            >
                        <ReactPlaceholder
                                    color="#CDCDCD"
                                    delay={0}
                                    ready={false}
                                    rows={3}
                                    showLoadingAnimation={true}
                                    type="text"
                        >
                                    No description
                        </ReactPlaceholder>
            </div>
</div>,
        ],
        "className": "ToolCard",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="ToolCardIcon"
>
                <AppIconMini
                                backgroundColor="#264D8C"
                                iconUrl="https://jumpto365.com/resources/images/app/Word.png"
                                name="word"
                                onClick={[Function]}
                />
</div>,
              <div
                className="ToolCardTitle"
                style={
                                Object {
                                                "lineHeight": "48px",
                                              }
                }
>
                <div
                                className="ms-fontSize-xl ms-fontWeight-semilight"
                />
</div>,
            ],
            "className": "ToolCardHeader",
            "style": Object {
              "backgroundColor": "#264D8C",
              "color": "#ffffff",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <AppIconMini
                  backgroundColor="#264D8C"
                  iconUrl="https://jumpto365.com/resources/images/app/Word.png"
                  name="word"
                  onClick={[Function]}
/>,
                "className": "ToolCardIcon",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "backgroundColor": "#264D8C",
                  "iconUrl": "https://jumpto365.com/resources/images/app/Word.png",
                  "name": "word",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="ms-fontSize-xl ms-fontWeight-semilight"
/>,
                "className": "ToolCardTitle",
                "style": Object {
                  "lineHeight": "48px",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "ms-fontSize-xl ms-fontWeight-semilight",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="ms-font-m"
>
              <ReactPlaceholder
                            color="#CDCDCD"
                            delay={0}
                            ready={false}
                            rows={3}
                            showLoadingAnimation={true}
                            type="text"
              >
                            No description
              </ReactPlaceholder>
</div>,
            "className": "ToolCardDetails",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ReactPlaceholder
                color="#CDCDCD"
                delay={0}
                ready={false}
                rows={3}
                showLoadingAnimation={true}
                type="text"
>
                No description
</ReactPlaceholder>,
              "className": "ms-font-m",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "No description",
                "color": "#CDCDCD",
                "delay": 0,
                "ready": false,
                "rows": 3,
                "showLoadingAnimation": true,
                "type": "text",
              },
              "ref": null,
              "rendered": "No description",
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
